.github/workflows/test-docker-build.yaml

```diff
@@ -0,0 +1,68 @@
+# This is a GitHub workflow defining a set of jobs with a set of steps.
+# ref: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions
+#
+name: Test docker multiarch build
+
+# Trigger the workflow's on all PRs and pushes so that other contributors can
+# run tests in their own forks. Avoid triggering these tests on changes to
+# documentation only changes.
+on:
+  pull_request:
+    paths-ignore:
+      - "doc/**"
+      - "**/test-docs.yaml"
+      - "**.md"
+      - "**/schema.yaml"
+  push:
+    paths-ignore:
+      - "doc/**"
+      - "**/test-docs.yaml"
+      - "**.md"
+      - "**/schema.yaml"
+    branches-ignore:
+      - "dependabot/**"
+  workflow_dispatch:
+
+jobs:
+  # TODO: this is just a quick test to check the arm64 docker images
+  # this should instead be done using chartpress
+  # Based on
+  # https://github.com/docker/build-push-action/blob/v2.3.0/docs/advanced/local-registry.md
+  # https://github.com/docker/build-push-action/blob/v2.3.0/docs/advanced/multi-platform.md
+  build_images:
+    runs-on: ubuntu-20.04
+    strategy:
+      # Keep running even if one variation of the job fails
+      fail-fast: false
+      matrix:
+        image:
+          - hub
+          - image-awaiter
+          - network-tools
+          - secret-sync
+          # - singleuser-sample
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      # https://github.com/docker/login-action/tree/v1.8.0#github-container-registry
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push to local registry
+        uses: docker/build-push-action@v2
+        with:
+          context: images/${{ matrix.image }}
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: ghcr.io/${{ github.repository_owner }}/z2jh-${{ matrix.image }}:dev
```

Comment:
Ohhh wow how nice and clean these steps are.

We have some complexity if we want to use this instead of build args and a dynamic tag based on chartpress.yaml and git inspection logic in chartpress. This looks great though.

Reply:
This is just a quick and easy way to expose the built images for download. I should have a chartpress PR to replace some of this soon :smile_cat: